Snipe improvements

DhanrajHira opened the pull request:
Since we keep a rolling record of messages deleted in the last n seconds, repeated snipes should not return the same messages.

Also added profile picture and the user's display name to the snipe embed,

Remember to squash and merge.

DhanrajHira committed:
Fix race condition in starboard.

DhanrajHira committed:
Better code formatting

DhanrajHira committed:
Merge branch 'master' of github.com:DhanrajHira/sherp2.0

DhanrajHira committed:
Repeated snipes don't return duplicate messages.

DhanrajHira committed:
Naming fix.

DhanrajHira committed:
Use discard instead of remove while deleting sniped messages.

Sooraj-beep commented on cogs/starboard.py:
```diff
@@ -98,12 +100,13 @@ async def _build_embeds(self, msg: discord.Message) -> List[discord.Embed]:
-            self._get_title(react), embeds=embeds, view=open_msg_view
-        )
-        self.starboard_msgs[react.message.id] = msg.id
+        async with self.__lock:
```
Was this meant to be included in a separate PR? I'm not sure what this change has to do with this PR

DhanrajHira replied:
You are correct this should not be part of this PR. I'll fix it.

DhanrajHira committed:
Merge branch 'master' of github.com:DhanrajHira/sherp2.0

DhanrajHira committed:
Merge branch 'master' into snipe-improvements

DhanrajHira committed:
Revert incorrect starboard changes.

DhanrajHira commented:
Fixed the SNAFU here. Please merge this before #77 .

DhanrajHira committed:
Python is not C++

DhanrajHira committed:
Actually do the thing this PR is meant for.

DhanrajHira committed:
Remove redundant book-keeping

DhanrajHira committed:
Change to dict to maintain insertion order.

This let's us ensure that the sniped messages are sent in the order they were deleted.

Sooraj-beep reviewed (approved):
LGTM, nice work on getting pfps showing!
